main.py

```diff
@@ -21,7 +21,60 @@ def parse(self, response):
         # open_in_browser(response)
 
     def parseResults(self, response):
-        test = response.xpath("//div[@id='ctl00_PlaceHolderMain_RecordSearchResultInfo_noDataMessageForSearchResultList_messageBar']")
-        test2 = response.xpath("//table[@id='ctl00_PlaceHolderMain_dgvPermitList_gdvPermitList']")
-        print("RESULTS", test, test2)
-        open_in_browser(response)
+        results_empty = response.xpath("//div[@id='ctl00_PlaceHolderMain_RecordSearchResultInfo_noDataMessageForSearchResultList_messageBar']").get()
+        records_rows = response.xpath("//table[@id='ctl00_PlaceHolderMain_dgvPermitList_gdvPermitList']/tr")
+
+        # Make sure that the page format is as expected.
+        # we expect that there's either no records with a message explaining that, or there are records without a message.
+        # If that's not true, it's likely that the page format has changed and this spider needs to be updated.
+        if results_empty:
+            assert len(records_rows) == 0, "Expected no records, but records were found"
+        else:
+            assert len(records_rows) > 0, "Expected records, but none were found"
+
+        # if there are records, extract them
+        if records_rows:
+            records = self.extractRecords(response, records_rows)
+            for record in records:
+                print(record)
```

Comment:
Do you anticipate extracting a function here in the future to either pretty-print or produce a file?

Reply:
Yeah, at this point whether it's printing to stdout, saving to a file, or saving to a global variable or something. I guess that'll somewhat depend on the hosting if we have access to persistent files

Reply:
I don't believe we'll need to persistently save these - they'll be used to send an email. The print statement is adequate as we build out the functionality.